client/rpcserver/types.go

```diff
@@ -289,13 +291,8 @@ func parseCancelArgs(params *RawParams) (*cancelForm, error) {
 	if len(id) != orderIdLen {
 		return nil, fmt.Errorf("%w: orderID has incorrect length", errArgs)
 	}
-	// We can only check up to 64 bits at a time. Break the 256 bit hex
-	// into 4 sections and check each.
-	div := orderIdLen / 4
-	for i := 0; i < 4; i++ {
-		if _, err := checkUIntArg(id[i*div:(i+1)*div], "orderID", 16, 64); err != nil {
-			return nil, fmt.Errorf("%w: invalid order id hex", errArgs)
-		}
+	if _, err := hex.DecodeString(id); err != nil {
+		return nil, fmt.Errorf("%w: invalid order id hex", errArgs)
```

Comment:
Wow, `%w` is great: https://blog.golang.org/go1.13-errors#TOC_3.3.  I didn't know about this addition to the language.